prime-router/src/main/kotlin/cli/tests/AuthTests.kt

```diff
@@ -551,6 +552,71 @@ class Server2ServerAuthTests : CoolTest() {
         settingsEnv = environment
         passed = passed and doEcAndRsaEcKeyTests(environment)
         passed = passed and doServer2ServerEndpointAuthTests(environment)
+        passed = passed and doApiKeyTests(environment)
+        return passed
+    }
+
+    private fun doApiKeyTests(environment: Environment): Boolean {
+        ugly("Starting $name test of adding server 2 server api keys")
+        var passed = true
+        val adminScope = "${org.name}.*.admin"
+        val mySenderName = "temporary_sender_auth_test"
+        val sender = createNewSenderForExistingOrg(mySenderName, org.name)
+        val kid = "adminkey"
+        saveServer2ServerKey(sender, end2EndExampleECPublicKeyStr, "adminkey", adminScope)
+        val (httpStatusGetToken, responseGetToken) = getServer2ServerAccessTok(
+            sender,
+            environment,
+            end2EndExampleECPrivateKeyStr,
+            kid,
+            adminScope
+        )
+        if (httpStatusGetToken == 200) {
+            good("Successfully got access token for ${org.name}")
+        } else {
+            bad("Failed to get access token for ${org.name}, response was $responseGetToken")
+            passed = false
+        }
+
+        val accessToken = jacksonObjectMapper().readTree(responseGetToken).get("access_token").textValue()
+        val headers = mutableListOf<Pair<String, String>>()
+        val clientStr = org.name
+        headers.add("client" to clientStr)
+        headers.add("authorization" to "Bearer $accessToken")
+        val postUrl =
+            "${environment.url}/api/settings/organizations/${org.name}/" +
+                "public-keys?scope=${org.name}.*.report&kid=${org.name}.report"
+        val (httpStatusPostKey, postKeyResponse) = HttpUtilities.postHttp(
+            postUrl,
+            end2EndExampleRSAPublicKeyStr.toByteArray(),
+            headers
+        )
+
+        if (httpStatusPostKey == 200) {
+            good("Successfully added key to ${org.name}")
+        } else {
+            bad("Failed to add key to ${org.name}, response was $postKeyResponse")
+            passed = false
+        }
+
+        val deleteUrl = environment.url.toString() +
+            "/api/settings/organizations/${org.name}/public-keys/" +
+            URLEncoder.encode("${org.name}.*.report", "utf-8") +
+            "/" +
+            URLEncoder.encode("${org.name}.report", "utf-8")
+        val (httpStatusDeleteKey, deleteKeyResponse) = HttpUtilities.deleteHttp(
+            deleteUrl,
+            byteArrayOf(),
+            headers
+        )
+
+        if (httpStatusDeleteKey == 200) {
+            good("Successfully removed key from ${org.name}")
+        } else {
+            bad("Failed to remove key from ${org.name}, response was $deleteKeyResponse")
+            passed = false
+        }
+
         return passed
```

Comment:
Should we do a call to get keys and make sure it is gone and we did not get an erroneous 200?

Reply:
Yep, that's a good call